zdm/energetics.py

```diff
@@ -10,20 +10,236 @@
+        truth[i] = mpmath.gammainc(gamma, a=av)
+    
+    # sets up for plotting results
+    from matplotlib import pyplot as plt
+    plt.figure()
+    ax1=plt.gca()
+    plt.plot(newavals,truth,label='truth')
+    plt.xlabel('a')
+    plt.ylabel('cumulative gamma function')
+    plt.xscale('log')
+    plt.yscale('log')
+    
+    # sets up for plotting differences between truth and interpolation
+    plt.figure()
+    ax2=plt.gca()
+    plt.xlabel('a')
+    plt.ylabel('|interpolation - truth|')
+    plt.xscale('log')
+    plt.yscale('log')
+    
+    # sets up for plotting relative differences between truth and interpolation
+    plt.figure()
+    ax3=plt.gca()
+    plt.xlabel('a')
+    plt.ylabel('|interpolation - truth|/truth')
+    plt.xscale('log')
+    plt.yscale('log')
+    
+    # now use different spline methods to evaluate
+    # standard method: cubic, linear
+    
+    SplineLog=False
+    init_igamma_splines([gamma],reinit=True,k=3)
+    result = interpolate.splev(newavals, igamma_splines[gamma])
+    ax1.plot(newavals,result,label='cubic spline, linear')
+    diff = np.abs(result - truth)
+    ax2.plot(newavals,diff,label='cubic spline, linear')
+    rdiff = diff/truth
+    ax3.plot(newavals,rdiff,label='cubic spline, linear')
+    
+    # now use different spline methods to evaluate
+    # standard method: cubic, linear
+    init_igamma_splines([gamma],reinit=True,k=1)
+    result = interpolate.splev(newavals, igamma_splines[gamma])
+    ax1.plot(newavals,result,label='linear spline, linear')
+    diff = np.abs(result - truth)
+    ax2.plot(newavals,diff,label='linear spline, linear')
+    rdiff = diff/truth
+    ax3.plot(newavals,rdiff,label='linear spline, linear')
+    
+    
+    SplineLog=True
+    init_igamma_splines([gamma],reinit=True,k=3)
+    result = 10**interpolate.splev(lnewavals, igamma_splines[gamma])
+    ax1.plot(newavals,result,label='cubic spline, log')
+    diff = np.abs(result - truth)
+    ax2.plot(newavals,diff,label='cubic spline, log')
+    rdiff = diff/truth
+    ax3.plot(newavals,rdiff,label='cubic spline, log')
+    
+    
+    # now use different spline methods to evaluate
+    # standard method: cubic, linear
+    init_igamma_splines([gamma],reinit=True,k=1)
+    result = 10**interpolate.splev(lnewavals, igamma_splines[gamma])
+    ax1.plot(newavals,result,label='linear spline, log')
+    diff = np.abs(result - truth)
+    ax2.plot(newavals,diff,label='linear spline, log')
+    rdiff = diff/truth
+    ax3.plot(newavals,rdiff,label='linear spline, log')
+    
+    plt.sca(ax1)
+    plt.legend()
+    plt.tight_layout()
+    plt.savefig('spline_test.pdf')
+    plt.close()
+    
+    plt.sca(ax2)
+    plt.legend()
+    plt.tight_layout()
+    plt.savefig('spline_errors.pdf')
+    
+    plt.sca(ax3)
+    plt.legend()
+    plt.ylim(1e-14,1)
+    plt.tight_layout()
+    plt.savefig('spline_rel_errors.pdf')
+    plt.close()
+    
 
+def time_splines(gamma = -1.5837, Ntimetest = 100000, Nreps=100):
```

Comment:
probably best in tests/ too

Reply:
Done! In same script

Reply:
(in theory, I could split this into a performance check file, but too lazy, since we haven't really paid attention to performance checks in any systematic way - should re-do these at some point)